packages/py-api/py_api/controllers/hackathon_participants_controller.py

```diff
@@ -145,3 +145,51 @@ def add_participant(participant: NewParticipant) -> JSONResponse:
             content={"message": "The participant was successfully added!"},
             status_code=200,
         )
+
+    @classmethod
+    def assign_random_team_to_participant(cls, object_id: str) -> JSONResponse:
+        # Fetch the teams of type random
+        avaliable_teams = []
+        random_teams = TeamsUtilities.fetch_teams_by_condition(
+            {"team_type": "random"},
+        )
+
+        # Find the teams which can accept a new participant
+        for team in random_teams:
+            if not TeamsUtilities.team_has_reach_max_cap(team):
+                avaliable_teams.append(team)
```

Comment:
I would suggest that you avoid using this `avaliable_teams` variable and instead have all of the functionality encapsulated within this `if not` block. 

```python
if not available_team := TeamsUtilities.team_has_reach_max_cap(team):
    available_team = TeamUtilities.add_participant_to_team(available_team.team_name, object_id)
...
```